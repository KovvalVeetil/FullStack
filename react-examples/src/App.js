import React, {Component} from 'react'; 
import './App.css';

class App extends Component {
    static allowedChars = /[a-zA-Z0-9\- .:=;!%]/; 
    constructor(props) {
        super(props);
        this.state = {password:'', passwordComment:null};
        this.updatePassword = this.updatePassword.bind(this); 
        this.logIt = this.logIt.bind(this);
        }
    logIt(event) {
        event.preventDefault();
        console.log('submit button clicked with password ' + this.state.password); }
    updatePassword(event) {
        // drop unallowable characters
        const newPassword = event.target.value.split('').filter(x => x.match(App.allowedChars)).join('');
        
        let comment;
        if (newPassword.length < 6) {
            comment = <p>Minimum length is 6 characters</p>; 
            } 
        else if (newPassword.length < 10) {
            comment = <p>Weak Password &mdash; longer passwords are more secure</p>;} 
        else {
            comment = <p>Strong Password</p>;
            }
        this.setState({password: newPassword, passwordComment: comment}); 
    }
    render() { 
        return (
            <div>
            <h1>Registration Form</h1> 
            <form>
                Password&nbsp;
                <input type="password" value={this.state.password}
                    onChange={this.updatePassword}/>
                <br/>
                { this.state.passwordComment }
                <button onClick={this.logIt}>Register</button>
                </form> 
                </div>
        );
    }
}
export default App;