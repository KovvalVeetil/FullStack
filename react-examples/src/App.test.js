// App.test.js
import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';
import {Collatz} from './App';

// Creating snapshot test of rendered items
it('snapshots Collatz with test renderer', () => {
    const collatz = renderer.create(<Collatz number={ 13 } />)

    // possible component interaction here
    let tree = collatz.toJSON();
    expect(tree).toMatchSnapshot();
});