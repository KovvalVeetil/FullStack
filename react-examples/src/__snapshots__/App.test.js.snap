// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Collatz with test renderer 1`] = `
.c0 {
  font-weight: bold;
  font-size: 90%;
  padding: 2px;
  list-style-type: none;
}

<div
  className="Collatz"
>
  <h2>
    Collatz Sequence for 
    13
  </h2>
  <ul>
    <li
      className="c0"
      listStyleType="none"
    >
      13
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      40
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      20
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      10
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      5
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      16
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      8
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      4
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      2
    </li>
    <li
      className="c0"
      listStyleType="none"
    >
      1
    </li>
  </ul>
</div>
`;
